packages/amplication-git-utils/src/git/git-factory.ts

```diff
@@ -3,10 +3,16 @@ import { EnumGitProvider, GitProvider, GitProviderArgs } from "../types";
 import { GithubService } from "./github.service";
 
 export class GitFactory {
-  public static getProvider(gitProviderArgs: GitProviderArgs): GitProvider {
+  public static async getProvider(
+    gitProviderArgs: GitProviderArgs
+  ): Promise<GitProvider> {
+    let gitProvider: GitProvider;
+
     switch (gitProviderArgs.provider) {
       case EnumGitProvider.Github:
-        return new GithubService(gitProviderArgs);
+        gitProvider = new GithubService(gitProviderArgs);
+        await gitProvider.init();
```

Comment:
Why do we need the init function, its a class, there is a contractor

Reply:
Because we need to perform async operation in the constructor and we can't, so this is the approach we took

Reply:
Got that, thanks, by the way, as a lib writer I suggest adding check to every function that the init has been run before the function run